test/src/e2e_vm_tests/test_programs/should_pass/language/retd_small_array/src/main.sw

```diff
@@ -0,0 +1,6 @@
+script;
+
+fn main() -> [u32; 1] {
+    let x: [u32; 1] = [1];
```

Comment:
Maybe also add a test for a zero-length array. Or is that not even allowed (should it be?)?

Reply:
Good question.  Rust says empty arrays are OK so I guess Sway does too.